src/navigation/MainTabNavigator.tsx

```diff
@@ -9,101 +13,154 @@ import HomeStack from "./HomeStack"
+    return tabIcon
+  }
+
+  const HeartbeatIcon: FunctionComponent<TabBarIconProps> = ({ focused }) => {
+    const tabIcon = <TabIcon icon={TabBarIcons.Heartbeat} focused={focused} />
+    return tabIcon
+  }
+
+  const SettingsIcon: FunctionComponent<TabBarIconProps> = ({ focused }) => {
+    const tabIcon = <TabIcon icon={TabBarIcons.Gear} focused={focused} />
+    return tabIcon
+  }
+
+  interface TabIconProps extends TabBarIconProps {
+    icon: string
+  }
+
+  const TabIcon: FunctionComponent<TabIconProps> = ({ focused, icon }) => {
+    const iconSize = 22
+
     return (
       <SvgXml
         xml={icon}
         fill={focused ? Colors.primary.shade100 : Colors.neutral.shade50}
-        width={size}
-        height={size}
+        width={iconSize}
+        height={iconSize}
       />
     )
   }
 
-  interface TabBarIconProps {
-    focused: boolean
-    size: number
-  }
-
-  const HomeIcon: FunctionComponent<TabBarIconProps> = ({ focused, size }) => {
-    return <TabIcon icon={TabBarIcons.House} focused={focused} size={size} />
+  type Tab_ = "home" | "exposureHistory" | "settings" | "symptomHistory"
+  const tabs: Tab_[] = ["home", "exposureHistory", "settings"]
+  if (displaySymptomHistory) {
+    tabs.push("symptomHistory")
   }
 
-  const ExposureHistoryIcon: FunctionComponent<TabBarIconProps> = ({
-    focused,
-    size,
+  const TabBar: FunctionComponent<BottomTabBarProps> = ({
+    state,
+    descriptors,
+    navigation,
   }) => {
-    const tabIcon = (
-      <TabIcon icon={TabBarIcons.Exposure} focused={focused} size={size} />
-    )
-    return tabIcon
-  }
+    return (
+      <View
+        style={{
+          flexDirection: "row",
+          justifyContent: "center",
+          paddingBottom: insets.bottom + Spacing.xxSmall,
+          paddingTop: Spacing.xxSmall,
+          backgroundColor: Colors.background.primaryLight,
+          borderTopWidth: Outlines.hairline,
+          borderColor: Colors.neutral.shade10,
+        }}
+      >
+        {state.routes.map((route, index: number) => {
+          const { options } = descriptors[route.key]
 
-  const HeartbeatIcon: FunctionComponent<TabBarIconProps> = ({
-    focused,
-    size,
-  }) => {
-    const tabIcon = (
-      <TabIcon icon={TabBarIcons.Heartbeat} focused={focused} size={size} />
-    )
-    return tabIcon
-  }
+          const label = options.title
+          const icon = options.tabBarIcon
+          const isFocused = state.index === index
 
-  const SettingsIcon: FunctionComponent<TabBarIconProps> = ({
-    focused,
-    size,
-  }) => {
-    const tabIcon = (
-      <TabIcon icon={TabBarIcons.Gear} focused={focused} size={size} />
+          const handleOnPress = () => {
+            !isFocused && navigation.navigate(route.name)
+          }
+
+          const textColor = isFocused
+            ? Colors.primary.shade100
+            : Colors.neutral.shade100
+
+          if (icon === undefined) {
```

Comment:
this is unfortunate, i wish react-navigation had more strict typings. What do you think about putting the early returns towards the top of the function?